File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function lcm(a, b) {
    return a * b / gcd(a, b);
}

function gcd(a, b) {
    while (true) {
        const r = a % b;
        if (r === 0) return b;
        a = b;
        b = r;
    }
}

function solution(arr) {
    arr.sort((a, b) => a - b);
    var lcm_v = arr[0];
    for (let i = 1; i < arr.length; i++) {
        lcm_v = lcm(lcm_v, arr[i]);
    }
    return lcm_v;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
import math

def solution(arr):
    lcm = arr[0] * arr[1] // math.gcd(arr[0], arr[1])  # 초기화
    
    for i in range(2, len(arr)):
        lcm = lcm * arr[i] // math.gcd(lcm, arr[i])
        
    return lcm